File: samples/SimpleFullStack/DotnetCoreApi/copilot-sample.Api/Models/Dtos/AddCategoryDto.cs
namespace copilot_sample.Api.Models.Dtos
{
    public class AddCategoryDto
    {
        public string Name { get; set; } = null!;
        public string? Description { get; set; }
        public int? ParentCategoryID { get; set; }
    }
    public class UpdateCategoryDescriptionDto
    {
        public string? Description { get; set; }
    }
}

File: samples/SimpleFullStack/Java/src/main/java/com/copilot/sample/dto/AddCategoryDto.java
package com.copilot.sample.dto;

import jakarta.validation.constraints.NotBlank;

public class AddCategoryDto {
    
    @NotBlank(message = "Name is required")
    private String name;
    
    private String description;
    private Integer parentCategoryId;
    
    // Constructors
    public AddCategoryDto() {}
    
    public AddCategoryDto(String name, String description, Integer parentCategoryId) {
        this.name = name;
        this.description = description;
        this.parentCategoryId = parentCategoryId;
    }
    
    // Getters and Setters
    public String getName() {
        return name;
    }
    
    public void setName(String name) {
        this.name = name;
    }
    
    public String getDescription() {
        return description;
    }
    
    public void setDescription(String description) {
        this.description = description;
    }
    
    public Integer getParentCategoryId() {
        return parentCategoryId;
    }
    
    public void setParentCategoryId(Integer parentCategoryId) {
        this.parentCategoryId = parentCategoryId;
    }
}
